src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -436,6 +470,50 @@ impl HummockEventHandler {
                             error!("unable to send flush result: {:?}", e);
                         });
                     }
+
+                    HummockEvent::RegisterReadVersion {
+                        table_id,
+                        new_read_version_sender,
+                    } => {
+                        let pinned_version = self.pinned_version.load();
+                        let basic_read_version = Arc::new(RwLock::new(HummockReadVersion::new(
+                            (**pinned_version).clone(),
+                        )));
+
+                        let instance_id = self.instance_id_generator.fetch_add(1, Ordering::SeqCst);
+
+                        {
+                            let mut read_version_mapping_guard = self.read_version_mapping.write();
+
+                            read_version_mapping_guard
+                                .entry(table_id)
+                                .or_default()
+                                .insert(instance_id, basic_read_version.clone());
+                        }
+
+                        match new_read_version_sender.send((basic_read_version, instance_id)) {
```

Comment:
We should simply print the error out instead of panic, because it may happen the actor is dropped while being built.

Reply:
I am thinking that we should return an instance id guard, which will automatically send the destroy event when dropped, so that we don't have to carefully handle sending the drop event. Without this guard, if the task for running actor is dropped while the actor is being built, it will forget to send the destroy event if the read version is sent but the local hummock is not built yet.